python/cpp/translator.cc

```diff
@@ -307,14 +307,18 @@ namespace ctranslate2 {
         // If the lock is not acquired immediately it means the model is being used
         // in another thread and we can't unload it at this time.
         std::unique_lock lock(_mutex, std::try_to_lock);
-        if (!lock || !_model_is_loaded)
+        if (!lock)
           return;
+        
+        std::vector<std::shared_ptr<const models::Model>> loaded_models;
+        if (_model_is_loaded)
+          loaded_models = _pool->detach_models();
 
-        _cached_models = _pool->detach_models();
-        if (to_cpu)
-          move_cached_models(Device::CPU, std::vector<int>(_cached_models.size(), 0));
-        else
+        if (to_cpu && _cached_models.empty())
+          _cached_models = clone_models(Device::CPU, std::vector<int>(loaded_models.size(), 0), loaded_models);
```

Comment:
I think you should keep ``move_cached_models`` like before. The cached_model is saved in CPU so the old logic is good, no need to create duplicate variable ``loaded_models``.

Reply:
@minhthuc2502 , thank you for your suggestion 🙂
Defining a new variable `loaded_models` is done to avoid overriding `_cached_models` in case we already copied models from GPU to CPU in the past. If we will override `_cached_models` with `_pool->detach_models()`, we will need to move models from GPU to CPU again, which is what this PR tries to avoid.
I can use the original `move_cached_models` instead of the `clone_models` method I created. But anyway I will have to change the `move_cached_models` method, because it will have to move models from `loaded_models` or from `_pool->detach_models()` and not from `_cached_models` (since `_cached_models` is empty or holds models in CPU - it will never hold models in GPU)

Reply:
Look good to me now! Thank you